awx/main/analytics/core.py

```diff
@@ -370,25 +371,31 @@ def ship(path):
     rh_user = getattr(settings, 'REDHAT_USERNAME', None)
     rh_password = getattr(settings, 'REDHAT_PASSWORD', None)
 
-    if not rh_user or not rh_password:
-        logger.info('REDHAT_USERNAME and REDHAT_PASSWORD are not set, using SUBSCRIPTIONS_USERNAME and SUBSCRIPTIONS_PASSWORD')
-        rh_user = getattr(settings, 'SUBSCRIPTIONS_USERNAME', None)
-        rh_password = getattr(settings, 'SUBSCRIPTIONS_PASSWORD', None)
-
-    if not rh_user:
-        logger.error('REDHAT_USERNAME and SUBSCRIPTIONS_USERNAME are not set')
-        return False
-    if not rh_password:
-        logger.error('REDHAT_PASSWORD and SUBSCRIPTIONS_USERNAME are not set')
-        return False
-
     with open(path, 'rb') as f:
         files = {'file': (os.path.basename(path), f, settings.INSIGHTS_AGENT_MIME)}
         s = requests.Session()
         s.headers = get_awx_http_client_headers()
         s.headers.pop('Content-Type')
         with set_environ(**settings.AWX_TASK_ENV):
-            response = s.post(url, files=files, verify=settings.INSIGHTS_CERT_PATH, auth=(rh_user, rh_password), headers=s.headers, timeout=(31, 31))
+            if rh_user and rh_password:
+                try:
+                    client = OIDCClient(rh_user, rh_password, url, ['api.console'])
```

Comment:
the 3rd parameter to OICClient is token_url, not the url we wish to make a request from. This can be hardcoded like we did here https://github.com/ansible/awx/blob/4487f2afa702e7740119ad50fc0d793abf017201/awx/playbooks/action_plugins/insights.py#L12